File: apps/agentsync/internal/agentvendor/service.go
package agentvendor

import "os/exec"

func IsInstalled(name AgentVendorName, getConfiguration ConfigurationPort) (bool, error) {
	configuration, err := getConfiguration(name)
	if err != nil {
		return false, err
	}
	_, err = exec.LookPath(configuration.PackageName)
	return err == nil, nil
}

File: apps/agentsync/src/agentsync/agent_vendor/service.py
import shutil

from agentsync.agent_vendor.models import AgentVendorName
from agentsync.agent_vendor.ports.configuration import AgentVendorConfigurationPort


def is_installed(
    vendor_name: AgentVendorName, configuration_port: AgentVendorConfigurationPort
) -> bool:
    configuration = configuration_port.get_configuration(vendor_name)
    return shutil.which(configuration.package_name) is not None
